### PR TITLE
Remove sender thread

### LINKED ISSUE #52: Consider switching threading models with regard to IO
Blocked by #50 

The way the current system works is that we have one thread that polls the Dragonfly API at a specified interval (60 seconds by default), and spawns tasks into the threadpool executor for each job returned by the API. Once these jobs are finished, they send their results over an mpsc channel. I think this adds an additional layer of complexity and necessitates an extra thread.

Instead, we should move over to a model where tasks will individually send their responses to the API as they finish. To avoid the reauthentication race-conditions we get with this models, this issue is blocked by #50

### PR DESCRIPTION
**Pull request by Robin5605**
Closes #52 

Removes the need for the "sender thread" and the mpsc channel by having each thread send it's results to the API directly, instead of over an mpsc channel. We don't need to worry about race conditions with reauthentication since that's been fixed.

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -80,61 +82,13 @@ fn main() -> Result<(), DragonflyError> {
         .with_env_filter(EnvFilter::from_default_env())
         .init();
     let client = Arc::new(DragonflyClient::new()?);
-    let (tx, rx) = mpsc::sync_channel(1024);
 
     // We spawn `n_jobs` threads using a threadpool for processing jobs, +1 more thread to send the
     // results. The main thread will handle requesting the jobs and submitting them to the threadpool.
```
Comment (jonathan-d-zhang) [inline src/main.rs:87]: will need to update this

**Review by jonathan-d-zhang — DISMISSED** [review]
Adjust README.md also and should be good

**Review by shenanigansd — DISMISSED** [review]
Not tested, :shipit:

**Review by shenanigansd — APPROVED** [review]
:shipit: 2, electric boogaloo

**Pull request merged**